src/main/java/org/passay/LengthComplexityRule.java

```diff
@@ -0,0 +1,407 @@
+/* See LICENSE for licensing and NOTICE for copyright. */
+package org.passay;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Rule for determining if a password contains the desired complexity for a certain length. In order to meet the
+ * criteria of this rule, passwords must meet any number of supplied rules.
+ *
+ * @author  Middleware Services
+ */
+public class LengthComplexityRule implements Rule
+{
+
+  /** Error code for insufficient number of characteristics. */
+  public static final String ERROR_CODE = "INSUFFICIENT_COMPLEXITY";
+
+  /** Rules to apply when checking a password. */
+  private final Map<Range, List<Rule>> rules = new HashMap<>();
+
+  /** Whether to report the details of each character rule failure. */
+  private boolean reportRuleFailures = true;
+
+
+  /**
+   * Sets the rules used by this rule.
+   *
+   * @param  range  of integers that the supplied rules apply to
+   * @param  l  list of rules
+   *
+   * @throws  IllegalArgumentException  if range is invalid or intersects with an existing range
+   */
+  public void setRules(final String range, final List<Rule> l)
+  {
+    final Range r = new Range(range);
+    for (Range existingRange : rules.keySet()) {
+      if (existingRange.intersects(r)) {
+        throw new IllegalArgumentException("Range " + r + " intersects existing range " + existingRange);
+      }
+    }
+    rules.put(r, l);
+  }
+
+
+  /**
+   * Sets the rules used by this rule.
+   *
+   * @param  range  of integers that the supplied rules apply to
+   * @param  r  list of rules
+   */
+  public void setRules(final String range, final Rule... r)
+  {
+    setRules(range, Arrays.asList(r));
+  }
+
+
+  /**
+   * Returns whether to add the rule result detail for each character rule that fails to validate to the rule result.
+   *
+   * @return  whether to add character rule result details
+   */
+  public boolean getReportRuleFailures()
+  {
+    return reportRuleFailures;
+  }
+
+
+  /**
+   * Sets whether to add the rule result detail for each character rule that fails to validate to the rule result.
+   *
+   * @param  b  whether to add character rule result details
+   */
+  public void setReportRuleFailures(final boolean b)
+  {
+    reportRuleFailures = b;
+  }
+
+
+  @Override
+  public RuleResult validate(final PasswordData passwordData)
+  {
+    final List<Rule> rulesByLength = getRulesByLength(passwordData.getPassword().length());
+    if (rulesByLength == null || rulesByLength.isEmpty()) {
```

Comment:
This reflects your comment on input validation. Currently the rule only checks for intersecting ranges. It does not (and probably can not) check that you've defined rules for [0, Integer.MAX_VALUE].

So this check must be done at runtime since rules in passay do not use an #initialize pattern.